_utils/firefox_download.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+
+from firefox_env_handler import IniHandler
+
+import os
+import sys
+import urllib.error
+import urllib.request
+
+
+class FirefoxDownload(object):
+
+    def __init__(self, config, out_dir='temp', clean=False):
+        """
+        TODO:
+        """
+        # Do some basic type checking on the `config` attribute.
+        if isinstance(config, IniHandler):
+            self.config = config
+
+        elif isinstance(config, str):
+            self.config = IniHandler()
+            self.config.load_os_config(config)
+
+        else:
+            sys.exit('FirefoxDownload: Unexpected config data type')
+
+        self.CACHE_FILE = 'cache.ini'
```

Comment:
probably shorter to just define this as a string constant at the top of the module one time and reference it that way throughout
CACHE_FILE = 'cache.ini'